charms/kserve-controller/tests/unit/test_charm.py

```diff
@@ -565,15 +566,23 @@ def test_restart_controller_service(harness, mocked_resource_handler, mocker):
                 "s3_secret_access_key": "test",
             },
             SECRETS_TEST_FILES,
-            '[{"apiVersion": "v1", "kind": "Secret", "metadata": {"name": "test", "annotations": {"serving.kserve.io/s3-endpoint": "test", "serving.kserve.io/s3-usehttps": "test", "serving.kserve.io/s3-region": "test", "serving.kserve.io/s3-useanoncredential": "test"}}, "type": "Opaque", "stringData": {"AWS_ACCESS_KEY_ID": "test", "AWS_SECRET_ACCESS_KEY": "test"}}]',  # noqa: E501
+            [
+                KubernetesManifest(
+                    manifest_content='apiVersion: v1\nkind: Secret\nmetadata:\n  name: test\n  annotations:\n     serving.kserve.io/s3-endpoint: "test"\n     serving.kserve.io/s3-usehttps: "test"\n     serving.kserve.io/s3-region: "test"\n     serving.kserve.io/s3-useanoncredential: "test"\ntype: Opaque\nstringData: # use `stringData` for raw credential string or `data` for base64 encoded string\n  AWS_ACCESS_KEY_ID: test\n  AWS_SECRET_ACCESS_KEY: test'
+                )
+            ],
         ),
         (
             {
                 "svc_account_name": "test",
                 "secret_name": "test",
             },
             SERVICE_ACCOUNTS_TEST_FILES,
-            '[{"apiVersion": "v1", "kind": "ServiceAccount", "metadata": {"name": "test"}, "secrets": [{"name": "test"}]}]',
+            [
+                KubernetesManifest(
+                    manifest_content="apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: test\nsecrets:\n- name: test"
+                )
+            ],
         ),
     ],
 )
```

Comment:
We could move the parameters to a separate file like we do in [test_seldon_servers](https://github.com/canonical/seldon-core-operator/blob/main/tests/integration/test_seldon_servers.py#L99-L108) in order to make this tests file more readable. WDYT?